### PR TITLE
PD-35412: Central Listener -  need to add Stripe Connected Account to the Payment Method - for Token Migration

### PR DESCRIPTION
**Pull request by stanleyrad1**
## DESCRIBE YOUR CHANGES

- Adds new logic to add metadata for customer and payment method
- Adds classes to communicate with Stripe

## TASK LINK

[PD-35412](https://togetherwork.atlassian.net/browse/PD-35412)

## DEPENDENT TASK (optional)

[Link to the PD ticket here]

## MANUAL TEST COVERAGE

[Explain how you tested your changes and verify that they work]

## QUICK CHECKLIST

-   [x] Self-reviewed my own code
-   [x] Added the corresponding test classes.
-   [x] Commented my code in hard-to-understand areas.
-   [x] New and existing test classes passed with my changes.

[PD-35412]: https://togetherwork.atlassian.net/browse/PD-35412?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by peterminetree** [comment]
Maybe disable triggers. Otherwise lgtm

**Review by peterminetree — APPROVED** [review]

File: force-app/main/default/classes/StripeAPIService.cls [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        true => new StripeCredentials(true),
+        false => new StripeCredentials(false)
+    };
+
+    public enum TransactionType {
+        UPDATE_PAYMENT_METHOD,
+        UPDATE_CUSTOMER
+    }
+
+
```
Comment (cesarParra) [inline force-app/main/default/classes/StripeAPIService.cls]: very minor: Remove extra blank space?

File: force-app/main/default/classes/StripeAPIService.cls [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    /**
+     * @description Map of TransactionType to the resource portion of the endpoint uri for
+     * the api request
+     */
+    private static final Map<TransactionType, String> resources = new Map<TransactionType, String> {
```
Comment (cesarParra) [inline force-app/main/default/classes/StripeAPIService.cls]: Minor: All caps for constant variable?

**Review by cesarParra — APPROVED** [review]
Looks great!

**Review by cesarParra — APPROVED** [review]

**Pull request merged**